analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -200,6 +215,21 @@ private static ProgramState ProcessAddMethod(ProgramState state, IMethodSymbol m
             {
                 state = state.SetSymbolConstraint(symbol, CollectionConstraint.NotEmpty);
             }
+            return state;
+        }
+        return null;
+    }
+
+    private static ProgramState ProcessRemoveMethod(ProgramState state, IMethodSymbol method, IOperation instance)
+    {
+        if (RemoveMethods.Contains(method.Name))
+        {
+            var value = (state[instance] ?? SymbolicValue.Empty).WithoutConstraint(CollectionConstraint.NotEmpty);
```

Comment:
this is the operation symbol without the constraint, right?
Took me some time to get what value is. Maybe a rename would help here.

Reply:
This line removes a specific constraint. I want to add convenience methods for this to `ProgramState` but not in this PR.